src/Features/Core/Portable/Diagnostics/IDiagnosticService.cs

```diff
@@ -25,11 +26,12 @@ internal interface IDiagnosticService
         /// <summary>
         /// Get current diagnostics stored in IDiagnosticUpdateSource
         /// </summary>
-        IEnumerable<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId projectId, DocumentId documentId, object id, bool includeSuppressedDiagnostics, CancellationToken cancellationToken);
+        ImmutableArray<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, CancellationToken cancellationToken);
```

Comment:
We should make sure none of the internal IVT clients use the changed APIs, or we might have to do a co-ordinated insertion.

Reply:
Actually, given just the return type is changing, we should probably be just fine.